src/devices/Bmx280/BmxBase.cs

```diff
@@ -305,24 +307,87 @@ private long CompensatePressure(int adcPressure)
             //Formula from the datasheet
             //The pressure is calculated using the compensation formula in the BMP280 datasheet
             long var1 = TemperatureFine - 128000;
-            long var2 = var1 * var1 * (long)_calibrationData.DigP6;
-            var2 = var2 + ((var1 * (long)_calibrationData.DigP5) << 17);
-            var2 = var2 + ((long)_calibrationData.DigP4 << 35);
-            var1 = ((var1 * var1 * (long)_calibrationData.DigP3) >> 8) + ((var1 * (long)_calibrationData.DigP2) << 12);
-            var1 = (((((long)1 << 47) + var1)) * (long)_calibrationData.DigP1) >> 33;
+            long var2 = var1 * var1 * (long)CalibrationData.DigP6;
+            var2 = var2 + ((var1 * (long)CalibrationData.DigP5) << 17);
+            var2 = var2 + ((long)CalibrationData.DigP4 << 35);
+            var1 = ((var1 * var1 * (long)CalibrationData.DigP3) >> 8) + ((var1 * (long)CalibrationData.DigP2) << 12);
+            var1 = (((((long)1 << 47) + var1)) * (long)CalibrationData.DigP1) >> 33;
             if (var1 == 0)
             {
                 return 0; //Avoid exception caused by division by zero
             }
             //Perform calibration operations
             long p = 1048576 - adcPressure;
             p = (((p << 31) - var2) * 3125) / var1;
-            var1 = ((long)_calibrationData.DigP9 * (p >> 13) * (p >> 13)) >> 25;
-            var2 = ((long)_calibrationData.DigP8 * p) >> 19;
-            p = ((p + var1 + var2) >> 8) + ((long)_calibrationData.DigP7 << 4);
+            var1 = ((long)CalibrationData.DigP9 * (p >> 13) * (p >> 13)) >> 25;
+            var2 = ((long)CalibrationData.DigP8 * p) >> 19;
+            p = ((p + var1 + var2) >> 8) + ((long)CalibrationData.DigP7 << 4);
             return p;
         }
 
+        /// <summary>
+        /// When called, the device is reset using the complete power-on-reset procedure.
+        /// </summary>
+        public void Reset()
+        {
+            const byte resetCommand = 0xb6;
+            _i2cDevice.Write(new[] { (byte)Register.RESET, resetCommand });
+        }
+
+        /// <summary>
+        /// Get the current status of the device.
+        /// </summary>
+        /// <returns></returns>
+        public DeviceStatus ReadStatus()
+        {
+            var status = Read8BitsFromRegister((byte)Register.STATUS);
+
+            // Bit 3
+            var measuring = (status & 0x0000_1000) >> 3 == 1;
+            // Bit 0
+            var imUpdate = (status & 0x0000_0001) == 1;
+
+            return new DeviceStatus
+            {
+                ImageUpdating = imUpdate,
+                Measuring = measuring
+            };
+        }
+
+        /// <summary>
+        /// Sets the IIR filter mode.
```

Comment:
Do you mind explaining what IIR means?

Reply:
It's what the datasheet uses to refer to the internal logic that reduces noise at the cost of accuracy - Infinite impulse response (IIR). The datasheet talks about different recommended filter settings based on usage. The IIR circuits are only used for temperature and and pressure.